packages/drivers/driver-base/src/documentDeltaConnection.ts

```diff
@@ -318,7 +318,27 @@ export class DocumentDeltaConnection
 	/**
 	 * Disconnect from the websocket and close the websocket too.
 	 */
-	protected closeSocket(error: IAnyDriverError) {
+	private closeSocket(error: IAnyDriverError) {
+		if (this._disposed) {
+			// This would be rare situation due to complexity around socket emitting events.
+			this.logger.sendTelemetryEvent(
+				{
+					eventName: "SocketCloseOnDisposedConnection",
+					driverVersion,
+					details: JSON.stringify({
+						disposed: this._disposed,
+						socketConnected: this.socket.connected,
```

Comment:
>this.socket [](http://example.com/codeflow?start=23&length=11)

can this be undefined in case it was disposed ? Should we safeguard ? same for the trackedListeners below

Reply:
I added the safeguard. TrackedListeners however could not be undefined.